concent_api/core/tests/test_core_views.py

```diff
@@ -72,3 +74,81 @@ def test_send_should_return_http_400_if_task_id_already_use(self):
         response = self.client.post(reverse('core:send'), data = json.dumps(self.correct_data), content_type = 'application/json', concent_client_public_key = self.public_key, HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key)
 
         self.assertEqual(response.status_code, 400)  # pylint: disable=no-member
+
+
+class CoreViewReceiveTest(TestCase):
+
+    def setUp(self):
+        self.client = Client()
+        self.public_key = '85cZzVjahnRpUBwm0zlNnqTdYom1LF1P1WNShLg17cmhN2UssnPrCjHKTi5susO3wrr/q07eswumbL82b4HgOw=='
+        self.force_data = {
+            "type":      "MessageForceReportComputedTask",
+            "timestamp": 1510911047,
+            "message_task_to_compute": {
+                "type":               "MessageTaskToCompute",
+                "timestamp":          1510909200,
+                "task_id":            1,
+                "deadline":           1510915047
+            }
+        }
+        self.ack_data = {
+            "type":      "MessageAckReportComputedTask",
+            "timestamp": 1510911047,
+            "message_task_to_compute": {
+                "type":               "MessageTaskToCompute",
+                "timestamp":          1510909200,
+                "task_id":            1,
+                "deadline":           1510915047
+            }
+        }
+        self.reject_data = {
+            "type":         "MessageRejectReportComputedTask",
+            "timestamp":    1510911047,
+            "reason":       "cannot-compute-task",
+            "message_cannot_commpute_task": {
+                "type":         "MessageCannotComputeTask",
+                "timestamp":    1510909200,
+                "reason":       "provider-quit",
+                "task_id":      1
+            }
+        }
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_receive_should_accept_valid_message(self):
+        message_timestamp   = datetime.datetime.now(timezone.utc)
+        new_message         = Message(
```

Comment:
Run validations here (to get an early warning if you have a bug in your data) and create both objects in a single call.

See my remarks about `save_messages_to_database()`.